src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Lambda/LambdaEventHelpers.cs

```diff
@@ -130,27 +130,39 @@ public static void AddEventTypeAttributes(IAgent agent, ITransaction transaction
     }
 
     private const string NEWRELIC_TRACE_HEADER = "newrelic";
+    private const string W3C_TRACEPARENT_HEADER = "traceparent";
+    private const string W3C_TRACESTATE_HEADER = "tracestate";
+    private static readonly List<string> TracingKeys = new List<string> { NEWRELIC_TRACE_HEADER, W3C_TRACEPARENT_HEADER, W3C_TRACESTATE_HEADER };
+    private const string SQS_MSG_ATTR_VALUE_PREFIX = @"Value"":""";
 
-    // TODO: Also handle W3C trace context headers
     private static void TryParseSQSDistributedTraceHeaders(dynamic sqsEvent, ITransaction transaction)
     {
         // We can't pass anything dynamic to AcceptDTHeaders, so we have to copy the sqs
         // message attributes to a new <string,string> dict and then pass that to AcceptDTHeaders
         IDictionary<string, string> sqsHeaders = new Dictionary<string, string>();
 
         var record = sqsEvent.Records[0];
```

Comment:
Do we need to add some defensive code here in case `.Records` is null or empty?

Reply:
Good idea, done.